demo/start.sh

```diff
@@ -56,26 +62,32 @@ for port in 8080 8081 8082; do
 		        compress: false
 		PORTLISTENERS
 		)
-        echo "${listeners}" >> $DIR/etc/$port.config
+        echo "${listeners}" >> "$DIR"/etc/$port.config
 
         # Disable tls for the servers
-        printf '\n\n# Disable tls on the servers.' >> $DIR/etc/$port.config
-        echo '# DO NOT USE IN PRODUCTION' >> $DIR/etc/$port.config
-        echo 'use_insecure_ssl_client_just_for_testing_do_not_use: true' >> $DIR/etc/$port.config
-        echo 'federation_verify_certificates: false' >> $DIR/etc/$port.config
+        {
+          printf '\n\n# Disable tls on the servers.'
+          echo '# DO NOT USE IN PRODUCTION'
+          echo 'use_insecure_ssl_client_just_for_testing_do_not_use: true'
+          echo 'federation_verify_certificates: false'
+        } >> "$DIR"/etc/$port.config
 
         # Set tls paths
-        echo "tls_certificate_path: \"$DIR/etc/localhost:$https_port.tls.crt\"" >> $DIR/etc/$port.config
-        echo "tls_private_key_path: \"$DIR/etc/localhost:$https_port.tls.key\"" >> $DIR/etc/$port.config
+        {
+          echo "tls_certificate_path: \"$DIR/etc/localhost:$https_port.tls.crt\""
+          echo "tls_private_key_path: \"$DIR/etc/localhost:$https_port.tls.key\""
+        } >> "$DIR"/etc/$port.config
```

Comment:
what is the logic behind merging *some* of the `printf`/`echo` commands into `{ ... } ` blocks? If you're going to do that, why not just make one big `{ ... }` block?